cli/src/command.rs

```diff
@@ -168,34 +168,36 @@ pub fn run() -> Result<()> {
 				let role = config.role.clone();
 
 				let task_manager = match role {
-					Role::Light => service::build_light(config).map(|(task_manager, _)| task_manager)
-					.map_err(|e| sc_service::Error::Other(e.to_string())),
+					Role::Light => service::build_light(config).map(|(task_manager, _)| task_manager),
 					_ => service::build_full(
 						config,
 						service::IsCollator::No,
 						grandpa_pause,
 						jaeger_agent,
 					).map(|full| full.task_manager)
-					.map_err(|e| sc_service::Error::Other(e.to_string()) )
-				};
+				}.map_err(|e| Error::PolkadotService(e));
 				task_manager
-			}).map_err(|e| -> sc_cli::Error { e.into() })?)
-
+			})?)
 		},
 		Some(Subcommand::BuildSpec(cmd)) => {
 			let runner = cli.create_runner(cmd)?;
-			Ok(runner.sync_run(|config| cmd.run(config.chain_spec, config.network))?)
+			Ok(runner.sync_run(|config| {
+				cmd.run(config.chain_spec, config.network)
+			})?)
 		},
 		Some(Subcommand::CheckBlock(cmd)) => {
-			let runner = cli.create_runner(cmd)?;
+			let runner = cli.create_runner(cmd)
+				.map_err(Error::SubstrateCli)?;
 			let chain_spec = &runner.config().chain_spec;
 
 			set_default_ss58_version(chain_spec);
 
 			runner.async_run(|mut config| {
-				let (client, _, import_queue, task_manager) = service::new_chain_ops(&mut config, None)
-					.map_err(|e| sc_service::Error::Other(e.to_string()))?;
-				Ok((cmd.run(client, import_queue), task_manager))
+				let (client, _, import_queue, task_manager) = service::new_chain_ops(&mut config, None)?;
+				Ok((async move {
+					cmd.run(client, import_queue).await
+						.map_err(|e| Error::SubstrateCli(e))
+				}, task_manager))
```

Comment:
Why not https://docs.rs/futures/0.3.9/futures/future/trait.TryFutureExt.html#method.map_err?

I don't see any reason to use await here.

Reply:
That was my first approach, but it yielded resolution errors when importing `use futures::future::FutureExt` with no added benefit. Is there a document outlining perf delta between an `async move {}` block and future combinators?